flutter_secure_storage_web/lib/flutter_secure_storage_web.dart

```diff
@@ -70,50 +71,46 @@ class FlutterSecureStorageWeb extends FlutterSecureStoragePlatform {
   }) async {
     final map = <String, String>{};
     final prefix = "${options[_publicKey]!}.";
-    for (int j = 0; j < html.window.localStorage.length; j++) {
-      final entry = html.window.localStorage.entries.elementAt(j);
-      if (!entry.key.startsWith(prefix)) {
+    for (int j = 0; j < web.window.localStorage.length; j++) {
+      final key = web.window.localStorage.key(j) ?? "";
+      if (!key.startsWith(prefix)) {
         continue;
       }
 
-      final value = await _decryptValue(entry.value, options);
+      final value = await _decryptValue(web.window.localStorage.getItem(key), options);
 
       if (value == null) {
         continue;
       }
 
-      map[entry.key.substring(prefix.length)] = value;
+      map[key.substring(prefix.length)] = value;
     }
 
     return map;
   }
 
-  crypto.Algorithm _getAlgorithm(Uint8List iv) =>
-      crypto.Algorithm(name: 'AES-GCM', length: 256, iv: iv);
+  js_interop.JSObject _getAlgorithm(Uint8List iv) {
+    return {'name': 'AES-GCM', 'length': 256, 'iv': iv}.jsify()! as js_interop.JSObject;
+  }
 
-  Future<html.CryptoKey> _getEncryptionKey(
-    crypto.Algorithm algorithm,
+  Future<web.CryptoKey> _getEncryptionKey(
+    js_interop.JSObject algorithm,
     Map<String, String> options,
   ) async {
-    late html.CryptoKey encryptionKey;
+    late web.CryptoKey encryptionKey;
     final key = options[_publicKey]!;
 
-    if (html.window.localStorage.containsKey(key)) {
-      final jwk = base64Decode(html.window.localStorage[key]!);
+    if (web.window.localStorage.has(key)) {
+      final jwk = base64Decode(web.window.localStorage[key]!);
 
-      encryptionKey = await js_util.promiseToFuture<html.CryptoKey>(
-        crypto.importKey("raw", jwk, algorithm, false, ["encrypt", "decrypt"]),
-      );
+      encryptionKey = await crypto.importKey("raw", jwk.toJS, algorithm, false, ["encrypt".toJS, "decrypt".toJS].toJS).toDart;
```

Comment:
Creating a private extension for `List<String>` seemed to increase readability. Please consider it if you like.

```dart
extension on List<String> {
  JSArray<JSString> get toJS => [
        ...map((e) => e.toJS),
      ].toJS;
}
```